Tests/MoyaProviderIntegrationTests.swift

```diff
@@ -299,6 +299,30 @@ final class MoyaProviderIntegrationTests: QuickSpec {
                 }
             }
         }
+
+        // Resolves ValidationType not working with multipart upload #1590
+        describe("a provider performing a multipart upload with Alamofire validation") {
+            it("only allows status code 287") {
+                let formData = HTTPBin.createMultipartFormData()
+                let target = HTTPBin.validatedUploadMultipart(formData, nil)
+                let provider = MoyaProvider<HTTPBin>()
+                provider.request(target) { result in
+                    switch result {
+                    case .success(let response):
+                        let validCodes = target.validationType.statusCodes
+                        expect(validCodes).to(contain(response.statusCode))
+                    case .failure(let error):
+                        switch error {
+                        case .underlying(_, let response):
+                            let validCodes = target.validationType.statusCodes
+                            expect(validCodes).toNot(contain(response!.statusCode))
```

Comment:
Can we assume to always have a `response` here?

Reply:
I think our hands are tied on this one. Response validation is only done on Alamofire integration tests. So I can't stub and ensure a response 😕 This could be an improvement